vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/resources/META-INF/resources/frontend/comboBoxConnector.js

```diff
@@ -260,8 +260,11 @@ import { ComboBoxPlaceholder } from '@vaadin/combo-box/src/vaadin-combo-box-plac
                 }
             }), { once: true });
 
-            comboBox.addEventListener('value-changed', tryCatchWrapper((event) => {
-                if (comboBox._selectedKey && !event.detail.value) {
+            comboBox.addEventListener('selected-item-changed', tryCatchWrapper((event) => {
```

Comment:
The `_selectedKey` property is originally [set](https://github.com/vaadin/flow-components/blob/8dc595c13137613db970412956a38757272a7989/vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBox.java#L1671) in ComboBox (on requested range change). Because of that, something about updating the value in the connector feels wrong.

Would it make sense to turn it into a synchronized property and have it update on requested range change and value change. Something like
```java
getElement().setProperty("_selectedKey",
  getValue() != null ? getKeyMapper().key(getValue()) : "");
```

I think it would resolve this issue also because the property would update on selection change.

Reply:
Removed the logic from the connector and used `addValueChangeListener` instead.